### PR TITLE
pull leaf functions out of VisualCPPMangler

### PR DESCRIPTION
**Pull request by WalterBright**
The idea is to pull them out:

1. to reduce the length of VisualCPPMangler
2. functions are easier to understand when their inputs and outputs are through the front door

This is a cut&paste job.

### CODE REVIEW

File: compiler/src/dmd/cppmanglewin.d [hunk 1]
```diff
@@ -1301,3 +1124,188 @@ extern(D):
...
+    case CppOperator.Cast:
+	buf.writestring("?B");
+	return true;
+    case CppOperator.Assign:
+	symName = "?4";
+	return false;
+    case CppOperator.Eq:
+	symName = "?8";
+	return false;
+    case CppOperator.Index:
```
Comment (thewilsonator) [inline compiler/src/dmd/cppmanglewin.d:1192]: indentation is all off

File: compiler/src/dmd/cppmanglewin.d [hunk 2]
```diff
@@ -1301,3 +1124,188 @@ extern(D):
...
+	}
+    }
+    continue_template:
+    if (ti.tiargs.length == 1)
+    {
+	buf.writestring(symName);
+	return true;
+    }
+    firstTemplateArg = 1;
+    return false;
```
Comment (thewilsonator) [inline compiler/src/dmd/cppmanglewin.d:1264]: ditto

File: compiler/src/dmd/cppmanglewin.d [hunk 3]
```diff
@@ -1301,3 +1124,188 @@ extern(D):
...
+	buf.writeByte('A');
+	buf.writeByte('@');
+	return;
+    }
+    if (num <= 10) // 5 encoded as "4"
+    {
+	buf.writeByte(cast(char)(num - 1 + '0'));
+	return;
+    }
+    char[17] buff = void;
```
Comment (thewilsonator) [inline compiler/src/dmd/cppmanglewin.d:1282]: ditto

File: compiler/src/dmd/cppmanglewin.d (hunk 1 above)
Comment (dkorpel) [reply compiler/src/dmd/cppmanglewin.d:1192]: Yes, it looks correct when tabs are 8 columns, but when you configure GitHub to make them 4 columns, you'll see the difference. In any case, the tabs should be converted to spaces to be consistent with the rest here.

Comment (WalterBright) [reply compiler/src/dmd/cppmanglewin.d]: Looks like I forgot to detab them. I thought the style checker would complain about that. Oh well. Detabbed.

**Review by dkorpel — APPROVED** [review]

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**